### PR TITLE
Merrp merge (September 9th)

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
This is a merrp merge - DO NOT SQUASH!

## Description
- aarant#40 (discarded, as it was already part of the expansion)
- [feat: static OW pokemon now bob while walking in place (per `OW_MON_BOBBING`)](https://github.com/rh-hideout/pokeemerald-expansion/commit/839cf2e79012e0fc9159af5ab9e6a497e86bbfa4)
- [feat: added OW_MON_WANDER_WALK config option](https://github.com/rh-hideout/pokeemerald-expansion/commit/42d9f24c8472a67d742d9d9da106480c84514336)

Also:
- Restored the name of `OW_FOLLOWERS_BOBBING` back to `OW_MON_BOBBING`, as it's not just for following Pokémon anymore.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: include/config/overworld.h [hunk 1]
```diff
@@ -42,9 +42,10 @@
                                                   // Compressed gfx are incompatible with non-power-of-two sprite sizes:
                                                   // (You should not use 48x48 sprites/tables for compressed gfx)
                                                   // 16x32, 32x32, 64x64 etc are fine
+#define OW_MON_BOBBING                 TRUE       // If true, OW pokemon will bob up and down during their idle & walking animations
+#define OW_MON_WANDER_WALK             TRUE       // If true, OW pokemon with MOVEMENT_TYPE_WANDER will walk-in-place in between steps.
```
Comment (AlexOn1ine) [inline include/config/overworld.h:46]: can both of those be true at the same time?

Comment (kittenchilly) [reply include/config/overworld.h:46]: Yes, I believe if they are both true at the same time then it mimics what following pokemon do

**Comment by Bassoonian** [comment]
I'm personally conflicted on if the `OW_FOLLOWERS_BOBBING` rename should be in a PR that is otherwise a very strict sync

**Comment by AsparagusEduardo** [comment]
> I'm personally conflicted on if the `OW_FOLLOWERS_BOBBING` rename should be in a PR that is otherwise a very strict sync

Hmm... can I at least add a comment saying that it applies to both followers and other overworld mon?

**Comment by Bassoonian** [comment]
I would personally leave the sync PRs as "pure" as they can be and just make a follow-up PR for the config rename. That way, we also have the syncing in the sync section in the changelog and the rename (which is expansion-related and has nothing to do with merrp's branch) elsewhere.

**Comment by AsparagusEduardo** [comment]
> I would personally leave the sync PRs as "pure" as they can be and just make a follow-up PR for the config rename. That way, we also have the syncing in the sync section in the changelog and the rename (which is expansion-related and has nothing to do with merrp's branch) elsewhere.

Would brace adjustments also fall into these "impure" changes?

**Comment by Bassoonian** [comment]
They wouldn't imo because that is just plain conversion of something to match expansion (like making something actually build when it otherwise wouldn't), so I personally consider that a necessity. Renaming a config falls under a different category imo.

**Comment by AsparagusEduardo** [comment]
Got it. I'll force-push to undo the name change commit.

**Comment by AsparagusEduardo** [comment]
Ready!

File: src/event_object_movement.c [hunk 2]
```diff
@@ -5660,6 +5667,14 @@ bool8 MovementType_MoveInPlace_Step1(struct ObjectEvent *objectEvent, struct Spr
 {
     if (ObjectEventExecSingleMovementAction(objectEvent, sprite))
         sprite->sTypeFuncId = 0;
+    // similar to UpdateMonMoveInPlace
+    else if (
+        OW_FOLLOWERS_BOBBING == TRUE
+        && IS_OW_MON_OBJ(objectEvent)
+        && (sprite->data[3] & 7) == 2)
```
Comment (Bassoonian) [inline src/event_object_movement.c:5674]:
```suggestion
    else if (OW_FOLLOWERS_BOBBING == TRUE
          && IS_OW_MON_OBJ(objectEvent)
          && (sprite->data[3] & 7) == 2)
```

File: src/event_object_movement.c [hunk 3]
```diff
@@ -3445,12 +3454,20 @@ bool8 MovementType_WanderAround_Step2(struct ObjectEvent *objectEvent, struct Sp
...
     if (WaitForMovementDelay(sprite))
     {
+        // resets a mid-movement sprite
+        ClearObjectEventMovement(objectEvent, sprite);
         sprite->sTypeFuncId = 4;
         return TRUE;
+    } else if (
+        OW_MON_WANDER_WALK == TRUE
+        && IS_OW_MON_OBJ(objectEvent))
+    {
```
Comment (Bassoonian) [inline src/event_object_movement.c:3469]:
```suggestion
    }
    else if (OW_MON_WANDER_WALK == TRUE && IS_OW_MON_OBJ(objectEvent))
    {
```

**Review by Bassoonian — COMMENTED** [review]
Some minor style remarks

**Comment by AsparagusEduardo** [comment]
Ready!

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**